Assets/_BForBoss/_Character/Scripts/DashLinesEffectBehaviour.cs

```diff
@@ -0,0 +1,54 @@
+//-----------------------------------------------------------------------
+// <copyright file="DashLinesEffectBehaviour.cs" company="PERIGON GAMES">
+//     Copyright (c) PERIGON GAMES. 2020 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Perigon.Character
+{
+    [RequireComponent(typeof(RawImage))]
+    public class DashLinesEffectBehaviour : MonoBehaviour
+    {
+        private const float START_TIME_OFFSET = 10f;
+        private const string MATERIAL_ALPHA = "_alpha";
+        
+        [SerializeField] [Range(0,1)] float _lerpInStrength = 0.1f;
+        [SerializeField] [Range(0,1)] float _lerpOutStrength = 0.1f;
+        [SerializeField] float _duration = 0.1f;
+        private float _currentTime;
+        
+        [SerializeField] [Range(0,1)] float _maxAlpha = 1f;
+        private Material _speedLineMaterial;
+        
+        private void Start()
+        {
+            _speedLineMaterial = GetComponent<RawImage>().material;
+            _speedLineMaterial.SetFloat(MATERIAL_ALPHA,  0);
+            _currentTime = _duration + START_TIME_OFFSET;
```

Comment:
So if it has to be at least 10. 
What you could do, is setup a `[Range(10, something large)]` above the _duration. 

Or name it like `MINIMUM_START_TIME`.

Basically, do you want others to know it has to be 10f minimum, or just force the users without knowing why, that it has to be 10 seconds.

Reply:
Okay so after looking at this a bit more, i see that you want to make it higher than the duration, So that in the beginning it instantly lerps to 0? 
Instead of having a random dash play on start?

Reply:
> I've renamed it as START_TIME_OFFSET, but the idea isn't easy to communicate through the variable name. Basically the _currentTime has to be an amount (arbitrarily 10) above the _duration to ensure that the dash lines start the scene with an alpha of 0 (at the end of the effect). The key point is that _currentTime is GREATER (>) than _duration rather than being EQUAL to _duration. Thus it is offset by an arbitrary value.

The value just has to be any value above 0 to make sure that the _currentTime starts at the end of the animation. So should I be putting `[Range(0.1, something large)] private const float START_TIME_OFFSET = 0.1f;` ?

Reply:
> Okay so after looking at this a bit more, i see that you want to make it higher than the duration, So that in the beginning it instantly lerps to 0?

Yeah that's the idea!

Reply:
ahh i see. 
Hmm this might just be a design problem then. 

Let's say you don't count up, but you count down instead.  And then you check if it's `x =< 0`
And then on `Start()`, `_currentTime` would already be at 0.  

Would that remove the need for an `OFFSET`

Reply:
Yeah that's probably a more elegant solution 👍